Code refactor on UX Improvement in Google Account in Setup MC

ecgan opened the pull request:
### Changes proposed in this Pull Request:

Part of #1065. This PR is targeting the PR #1072 branch `feature/1065/google-account`.

This code refactor comes up from my discovery and work on PR #1072. The idea here is to simplify things, remove duplicated code, make things more straightforward, clearer and simpler to understand.

- I came up with a `ConnectGoogleAccountCard`. It is refactored out from `AuthorizeGoogleAccountCard`, so now both of them does not contain any conditional rendering with ternary operator.
- The components also accept different props. `ConnectGoogleAccountCard` only needs `disabled` props and does not need any email. `AuthorizeGoogleAccountCard` only needs `additionalScopeEmail` props and does not need `disabled` props.
- `ConnectGoogleAccountCard` and `AuthorizeGoogleAccountCard` is conditionally rendered in `GoogleAccountCard`.
- A new hook `useGoogleConnectFlow` is created to wrap the common shared logic used in `AuthorizeGoogleAccountCard`, `ConnectGoogleAccountCard` and `ConnectedGoogleAccountCard`. Note that if the logic changes in any one of the card component in the future, it may be wise to not use the hook.

### Screenshots:

None. Everything should still work the same as per PR #1072.

### Detailed test instructions:

Refer to PR #1072 with demo video.

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

>

ecgan committed:
Code refactor with ConnectGoogleAccountCard.

ecgan committed:
Code refactor with readMoreLink.

ecgan committed:
Remove unneeded isAskingScope in AuthorizeGoogleAccountCard.

ecgan committed:
Remove unneeded disabled prop.

ecgan committed:
Code refactor with useGoogleConnectFlow.

eason9487 commented on js/src/components/google-account-card/authorize-google-account-card.js:
```diff
@@ -7,116 +7,60 @@ import { createInterpolateElement } from '@wordpress/element';
  * Renders an AccountCard based on Google appearance for requesting Google authorization from user.
  *
  * @param {Object} props React props.
  * @param {string} [props.additionalScopeEmail] Specify the email to be requested additional scopes. Set this prop only if wants to request a partial oauth to Google.
```
`additionalScopeEmail` is changed to a required prop.

```suggestion
 * @param {string} props.additionalScopeEmail Specify the email to be requested additional permission scopes to Google.
```

eason9487 commented on js/src/components/google-account-card/authorize-google-account-card.js:
```diff
@@ -7,116 +7,60 @@ import { createInterpolateElement } from '@wordpress/element';
-);
-
 /**
  * Renders an AccountCard based on Google appearance for requesting Google authorization from user.
```
💅 
Since I named the component starting with **Authorize**, to align with Google OAuth 2.0's "authorization" term. When asking users for authorizing GLA app, it covers both basic access to account information (`openid` and `userinfo.email`), as well as additional optional permission scopes (`content`, `siteverification.verify_only` and `adwords`).

If want to separate the requesting of Google authorization into two actions in GLA, I would suggest avoiding the combination of **Connect** and **Authorize** to prevent the ambiguity that **Authorize** on Google auth perspective can conceptually cover **Connect** on GLA perspective as well. But this component is going to be used for requesting more permission to users only.  Or, at least update its JSDoc description for clarifying the component usage.

A relevant context - https://github.com/woocommerce/google-listings-and-ads/pull/1052#discussion_r733333274

ecgan replied:
Good point! I was thinking to rename the component to something like `RequestFullAccessGoogleAccountCard` to reflect the "Allow full access" button text in the UI and its responsibility, but I didn't do that because I thought **Authorize** word is fine enough. I will change the component name so that it is clearer and no ambiguity, and update the JSDoc description. 🙂  👍

ecgan replied:
Done in [`3f75f9a` (#1074)](https://github.com/woocommerce/google-listings-and-ads/pull/1074/commits/3f75f9a911e1c33654d6f9d7fffba7646f0e2e71).

eason9487 commented on js/src/components/google-account-card/connect-google-account-card.js:
```diff
@@ -0,0 +1,57 @@
+					isSecondary
+					disabled={ disabled }
+					loading={ loading || data }
+					eventName="gla_google_account_connect_button_click"
```
💅 
Maybe add [the `eventProps`](https://github.com/woocommerce/google-listings-and-ads/pull/1069/commits/c4761c141286e19c937f41e89036871021ad30d6#diff-985cbb1246d5a14d2eb12acaf614585f17c15b4d6062c1c224c959a6a781387dR119-R122) first to avoid missing it when dealing with merging conflicts, or missing it directly if no conflicts happened.

ecgan replied:
Done in [`65478e6` (#1074)](https://github.com/woocommerce/google-listings-and-ads/pull/1074/commits/65478e6823aa646e98ac46902759d085100671a4).

eason9487 commented on js/src/components/google-account-card/use-google-connect-flow.js:
```diff
@@ -0,0 +1,54 @@
+ *
+ * The `handleConnect` handler is meant to be used in button click handler. Upon button click, the handler will:
+ *
+ * 1. Call `fetchGoogleConnect` from `useGoogleAuthorization` hook to get the Google OAUth URL.
```
💅  A char case typo. 😂 

```suggestion
 * 1. Call `fetchGoogleConnect` from `useGoogleAuthorization` hook to get the Google OAuth URL.
```

eason9487 commented on js/src/components/google-account-card/use-google-connect-flow.js:
```diff
@@ -0,0 +1,54 @@
+ * @see https://developers.google.com/identity/protocols/oauth2/openid-connect#login-hint
+ * @return {Array} `[ handleConnect, fetchResult ]`
+ * 		- `handleConnect` is meant to be used as button click handler.
+ * 		- `fetchResult` is the same returned object from `useApiFetchCallback`.
```
💅 
Suggest avoiding naming a data payload object starting with a verb.

ecgan replied:
Done in [`035aceb` (#1074)](https://github.com/woocommerce/google-listings-and-ads/pull/1074/commits/035aceb35afea0ec0ad5771c9fc0ae89700187c9).

eason9487 reviewed (approved):
Tested locally and works correctly. Left some comments could be polished a bit. Since all comments are not related to logic, approved in advance.

eason9487 commented on js/src/components/google-account-card/authorize-google-account-card.js:
```diff
@@ -7,116 +7,60 @@ import { createInterpolateElement } from '@wordpress/element';
-					disabled={ disabled }
+					isDestructive
 					loading={ loading || data }
 					eventName="gla_google_account_connect_button_click"
```
Similar to https://github.com/woocommerce/google-listings-and-ads/pull/1074#discussion_r741671498. But I'm thinking maybe add `eventProps` here to force them to appear in conflict, so that we would not miss it. 😆

ecgan replied:
Done in [`65478e6` (#1074)](https://github.com/woocommerce/google-listings-and-ads/pull/1074/commits/65478e6823aa646e98ac46902759d085100671a4).

ecgan committed:
Fix JSDoc in AuthorizeGoogleAccountCard.

Co-authored-by: Eason <eason.su.tw@gmail.com>

ecgan committed:
Fix typo in useGoogleConnectFlow JSDoc.

Co-authored-by: Eason <eason.su.tw@gmail.com>

ecgan committed:
Merge branch 'feature/1065/google-account' into feature/1065/refactor-google-account

Conflicts:
	js/src/components/google-account-card/connected-google-account-card.js

ecgan committed:
Rename AuthorizeGoogleAccountCard to RequestFullAccessGoogleAccountCard.

ecgan committed:
Add eventProps for gla_google_account_connect_button_click event.

This is done to to better handle future merge conflict with this commit: https://github.com/woocommerce/google-listings-and-ads/pull/1069/commits/c4761c141286e19c937f41e89036871021ad30d6#diff-985cbb1246d5a14d2eb12acaf614585f17c15b4d6062c1c224c959a6a781387dR119-R122, as mentioned in https://github.com/woocommerce/google-listings-and-ads/pull/1074#discussion_r741671498.

ecgan committed:
Change useGoogleConnectFlow to take pageName parameter.

ecgan committed:
Avoid naming data object with a prefix that can be a verb.

ecgan commented:
I have addressed the PR review comments here. I will proceed to merge this PR when PR #1072 is done.
